src/Type/Definition/UnionType.php

```diff
@@ -98,6 +98,7 @@ public function getTypes(): array
                 );
             }
 
+            /** @var ObjectType|callable():ObjectType $type */
```

Comment:
Let's not. I would rather ignore an error if type inference works incorrectly, or actually fix what is wrong. `@var` permanently circumvents correct analysis.

Reply:
true. it somehow connect invoke $types without casting it to mixed